plugins/clipboard/plugin.js

```diff
@@ -368,7 +368,7 @@
 						editable_wrapper,
 						wrapper = new CKEDITOR.dom.element( 'div' );
 
-					wrapper.setHtml( data );
+					wrapper.setHtml( CKEDITOR.tools.htmlSafeByReview(data, 'We assume eventData is not injectable and is only used internally.') );
```

Comment:
Let's check if we can sanitize after the HTML is constructed in `transformPlainTextToHtml`. If the plain text contains a `<script>` will it execute?

Reply:
The `transformPlainTextToHtml` function uses htmlEncode and only decodes certain characters like whitespaces or tabs so that output should already be sanitized. It seems that the conditional that reaches transformPlainTextToHtml is not triggered for this scenario though.
I tried messing around with the testing scheme for this function and was able to get some sort of injection at least from the tests: 
![image](https://github.com/eozmen410/ckeditor4/assets/16466991/97b2bd91-8622-4679-be26-7c3c6b883d48)
It looks like `wrapper` the variable we run setHtml on is never attached to the page though, and is only used for some extra computations that will return to data on line 382: `data = editable_wrapper.getHtml().replace( /<br>$/i, '' );`